### DISCUSSION TITLE
"fetch is not defined" on an unrelated file

### DISCUSSION THREAD

**Discussion by sergiomartindev** [vitest-dev/vitest#2780] [Q&A] [answered]
I'm implementing Vitest in my project and I am facing a problem that escapes from my understandment. 

My vitest.config content is the following:

```
import { mergeConfig } from 'vite'
import { defineConfig } from 'vitest/config'
import viteConfig from './vite.config'

// https://vitest.dev/config/
export default mergeConfig(viteConfig, defineConfig({
  test: {
    include: ['**/tests/*.{test,spec}.{ts,mts,cts,jsx,tsx}'],
    globals: true,
    environment: "happy-dom",
  },
}))
```

At this moment, I have only one test file defined in my entire project.

When I run the `vitest --reporter verbose` command, I face this error message:

```
 FAIL  src/domains/app/tests/evaluate.utils.test.ts [ src/domains/app/tests/evaluate.utils.test.ts ]
ReferenceError: fetch is not defined
 ❯ src/domains/app/services/appConfig.service.ts:26:6
     24|   if (!appSettingsLoadPromise) {
     25|     appSettingsLoadPromise = new Promise(resolve => {
     26|       fetch('/appsettings.json')
       |      ^
```

I understand that Vitest shouldn't know what "fetch" is, but that "fetch" is being used in another file with no apparent connection with the *.test.ts file I want to test.

Any clue or idea about what is going on?

**Comment by sheremet-va** [comment, marked as answer]
You can see the whole module graph with `vitest --ui` to find what file imports it

**Reply by sergiomartindev** [reply]
Your reply was very useful and accurate.

Lost inside a myriad of imports, I could find the origin of the problem. There was a module that was calling a function that was performing the fetch. Something like this:

```
// blablabla.module
callTheFunctionThatWasCausingTroubles();

function a () {}

function b () {}

export default { a, b }
```
